packages/ai-jsx/src/lib/openai.tsx

```diff
@@ -370,6 +442,14 @@ export async function* OpenAIChatModel(
             },
           };
         case 'functionResponse':
+          if (message.element.props.id && !unmatchedIds.has(message.element.props.id)) {
+            return {
+              role: 'tool',
+              tool_call_id: message.element.props.id,
+              content: await render(message.element.props.children),
+            };
+          }
+
           return {
```

Comment:
Same here, can't we only have `tool` not both `tool` and `function`?

Reply:
Or will OAI require IDs for all `tools` but not `function`s?

Reply:
Actually, you said that if we use `functions` they will respond with `delta.function_call`, so this is probably more serious since we don't handle that anymore.

Reply:
> Or will OAI require IDs for all tools but not functions?

Yup exactly. OAI is strict w.r.t. to tools, but not w.r.t. functions.